lib/redox/connection.rb

```diff
@@ -10,19 +10,30 @@ class Connection
 
     format :json
 
-    def request(endpoint: DEFAULT_ENDPOINT, body: nil, headers: {}, auth: true)
+    def post(endpoint: DEFAULT_ENDPOINT, body: nil, headers: {}, auth: true)
       body    = body.to_json if body.is_a?(Hash)
-      headers = auth_header.merge(headers) if auth
+      headers = auth_header(auth_class(endpoint)).merge(headers) if auth
 
       self.class.post(endpoint, body: body, headers: headers)
     end
+    alias :request :post
+
+    def get(endpoint:, headers: {}, auth: true)
+      headers = auth_header(auth_class(endpoint)).merge(headers) if auth
+
+      self.class.get(endpoint, headers: headers)
+    end
 
     private
 
-    def auth_header
-      @auth ||= Authentication.new
+    def auth_class(endpoint)
```

Comment:
@khall 
👍  for adding additional authentication for FHIR/platform.
Please adjust for backwards compatibility of legacy authentication. We would plan to release the deprecation as a minor version update following semantic versioning.